wrappers/java/src/main/java/org/hyperledger/indy/sdk/agent/Agent.java

```diff
@@ -0,0 +1,329 @@
+package org.hyperledger.indy.sdk.agent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+import org.hyperledger.indy.sdk.LibSovrin;
+import org.hyperledger.indy.sdk.SovrinException;
+import org.hyperledger.indy.sdk.SovrinJava;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentAddIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseConnectionResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseListenerResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentConnectResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentListenResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentRemoveIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentSendResult;
+import org.hyperledger.indy.sdk.pool.Pool;
+import org.hyperledger.indy.sdk.wallet.Wallet;
+
+import com.sun.jna.Callback;
+
+/**
+ * agent.rs API
+ */
+public class Agent extends SovrinJava.API {
+
+	private Agent() {
+
+	}
+
+	/*
+	 * STATIC METHODS
+	 */
+
+	public static Future<AgentConnectResult> agentConnect(
+			Pool pool,
+			Wallet wallet,
+			String senderDid,
+			String receiverDid,
+			Callback messageCb) throws SovrinException {
```

Comment:
see dcbd87f19300dfbf1795c4ee2c5762caf6914449 for initial commit to use observer pattern